devito/tools/threading.py

```diff
@@ -1,6 +1,26 @@
 import threading
 
-__all__ = ['sympy_mutex']
+__all__ = ['sympy_mutex', 'safe_dict_copy']
 
 
 sympy_mutex = threading.RLock()
+
+
+def safe_dict_copy(v):
```

Comment:
when/where is that an issue? Caching I see but why is it problematic now?

Reply:
PR was triggering this failure:
https://github.com/devitocodes/devito/runs/6078671948?check_suite_focus=true

Discussed a few alternatives with @tjb900 and @ggorman , eventually picked this one

Reply:
Ok. Weird never triggered before.